### PR TITLE
[BEAM-360] Adds a PTransform for Avro source and updates snippets.

### PR DESCRIPTION
**Pull request by chamikaramj**
Wrapping a custom source as a 'PTransform' is better than directly using the source using 'df.Read' since the 'PTransform' can be extended without breaking end-user code.

Updates the documentation of avroio module.

Adds 'PTransform' wrappers to custom sources and sinks in 'snippets.py'.

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @silviulica

File: sdks/python/apache_beam/examples/snippets/snippets.py [hunk 1]
```diff
@@ -698,17 +738,23 @@ def model_custom_sink(simplekv, KVs, final_table_name):
...
                   creates a table named 'table_name'. Returns a table object.
-              simplekv.write_to_table(table, access_token, key, value) -
+              simplekv.write_to_table(access_token, table, key, value) -
                   writes a key-value pair to the given table.
               simplekv.rename_table(access_token, old_name, new_name) -
                   renames the table named 'old_name' to 'new_name'.
     KVs: the set of key-value pairs to be written in the example pipeline.
-    final_table_name: the prefix of final set of tables to be created by the
-                      example pipeline.
+    final_table_name_no_ptransform: the prefix of final set of tables to be
```
Comment (silviulica) [inline sdks/python/apache_beam/examples/snippets/snippets.py]: I do not quite grasp these two args: no_ptransform and with_ptransform. Perhaps a more detailed comment in the function comment. I would also move the comment about the KV APIs for the simplekv argument to the top function comment.

File: sdks/python/apache_beam/io/avroio.py [hunk 2]
```diff
@@ -26,44 +26,71 @@
 from avro import schema
 
 from apache_beam.io import filebasedsource
+from apache_beam.io.iobase import Read
+from apache_beam.transforms import PTransform
+
+
+class ReadFromAvro(PTransform):
+  """A ``PTrasnfrom`` for reading avro files."""
```
Comment (silviulica) [inline sdks/python/apache_beam/io/avroio.py]: PTransform typo

File: sdks/python/apache_beam/io/avroio.py [hunk 3]
```diff
@@ -26,44 +26,71 @@
...
+
+    Each record of this ``PCollection`` will contain a single record read from a
+    source. Records that are of simple types will be mapped into corresponding
+    Python types. Records that are of Avro type 'RECORD' will be mapped to
+    Python dictionaries that comply with the schema contained in the Avro file
+    that contains those records. In this case, keys of each dictionary
+    will contain the corresponding field names and will be of type ``string``
+    while the values of the dictionary will be of the type defined in the
+    corresponding Avro schema.
+    For example, if schema of the Avro file is following.
```
Comment (silviulica) [inline sdks/python/apache_beam/io/avroio.py]: is the following:

**Comment by silviulica** [comment]
LGTM - had just a bunch of cosmetic changes.
Thanks Cham!

File: sdks/python/apache_beam/examples/snippets/snippets.py (hunk 1 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/examples/snippets/snippets.py]: Added more comments explaining this. Moved the comment describing the API to the top level.

File: sdks/python/apache_beam/io/avroio.py (hunk 2 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/avroio.py]: Done.

File: sdks/python/apache_beam/io/avroio.py (hunk 3 above)
Comment (chamikaramj) [inline sdks/python/apache_beam/io/avroio.py]: Done.

**Comment by chamikaramj** [comment]
R: @robertwb

**Comment by chamikaramj** [comment]
Thanks Dan.

**Pull request closed without merging**